routing/payment_lifecycle.go

```diff
@@ -247,7 +247,19 @@ func (p *paymentLifecycle) resumePayment() ([32]byte, *route.Route, error) {
 
 		// We found a route to try, launch a new shard.
 		attempt, outcome, err := shardHandler.launchShard(rt)
-		if err != nil {
+		switch err {
+		// We can get an already failed error in the case where we
+		// received a permanent failure while we were pathfinding for
+		// this shard.
+		case channeldb.ErrPaymentAlreadyFailed:
```

Comment:
I initially found it a bit weird to handle the error here and where the control tower is actually called. But looking at it more closely this might be best, since otherwise we would have to signal back to this loop whether to expect a result to be collected or not (re-purposing the `launchOutcome` would be an option).

Just saying there might exist a cleaner way to do this, but I'm sure this is correct :)

Reply:
Also considered this approach, ended up leaving it in here because this case only affects this call to `launchShard` because we're operating in the context of this loop and (I think?) we want to leave `SendToRoute` as is and error out on `ErrPaymentAlreadyFailed`?